src/@types/events.ts

```diff
@@ -0,0 +1,35 @@
+import { Action, SerializedError } from '@reduxjs/toolkit';
+import { ApplicationState } from '../store/types';
+
+export enum Event {
+    CREATE_ANNOTATION = 'annotationCreate',
+}
+
+export enum Status {
+    ERROR = 'error',
+    PENDING = 'pending',
+    SUCCESS = 'success',
+}
+
+export interface Metadata {
+    status: Status;
+}
+export interface ActionEvent {
+    annotation: object | undefined;
+    error: Error | undefined;
+    meta: Metadata;
+}
+
+export interface ThunkMeta {
+    arg: NewAnnotation;
```

Comment:
Will this support different types of async events that may have different arguments/payloads?